nemo_text_processing/text_normalization/vi/taggers/tokenize_and_classify.py

```diff
@@ -92,9 +93,15 @@ def __init__(
             fraction_graph = fraction.fst
             logger.debug(f"fraction: {time.time() - start_time: .2f}s -- {fraction_graph.num_states()} nodes")
 
+            start_time = time.time()
+            date = DateFst(cardinal=cardinal, deterministic=deterministic)
+            date_graph = date.fst
+            logger.debug(f"date: {time.time() - start_time: .2f}s -- {date_graph.num_states()} nodes")
+
             classify = (
                 pynutil.add_weight(whitelist_graph, 0.8)
                 | pynutil.add_weight(ordinal_graph, 0.81)
+                | pynutil.add_weight(date_graph, 0.83)
```

Comment:
As Simon said, this this level of granularity is a problem with the graph. If you truly need this we should look into simply training the weights.